libraries/AP_Proximity/AP_Proximity_MAV.cpp

```diff
@@ -226,11 +228,19 @@ void AP_Proximity_MAV::handle_obstacle_distance_3d_msg(const mavlink_message_t &
     Matrix3f body_to_ned;
     const bool database_ready = database_prepare_for_push(current_pos, body_to_ned);
 
-    const Vector3f obstacle(packet.x, packet.y, packet.z * -1.0f);
-    if (obstacle.length() < _distance_min || obstacle.length() > _distance_max || obstacle.is_zero()) {
+    const Vector3f obstacle_FRD(packet.x, packet.y, packet.z);
+    if (obstacle_FRD.length() < _distance_min || obstacle_FRD.length() > _distance_max || obstacle_FRD.is_zero()) {
```

Comment:
a small optimisation here would be to calculate obstacle_FRD.length just once.  It uses a sqrt so it does take a bit of CPU to calculate it.